openmls/src/group/core_group/apply_proposals.rs

```diff
@@ -151,7 +152,9 @@ impl CoreGroup {
         // Extract KeyPackages from proposals
         let mut invitation_list = Vec::new();
         for add_proposal in add_proposals {
-            let leaf_index = diff.add_leaf(add_proposal.key_package().clone(), backend.crypto())?;
+            let leaf_index = diff
+                .add_leaf(add_proposal.key_package().clone(), backend.crypto())
+                .map_err(|_| LibraryError::custom("Tree full: cannot add more members"))?;
```

Comment:
This is something that can actually happen if the group gets too large. Is there already a check somewhere else that would make this a `LibraryError`? (Same for the other use of `add_leaf()` in `create_commit`.